pulsar/producer_partition.go

```diff
@@ -520,7 +520,12 @@ func (p *partitionProducer) internalSend(request *sendRequest) {
 	}
 
 	// if msg is too large
-	if len(payload) > int(p._getConn().GetMaxMessageSize()) {
+	minMessageSize := len(payload)
+	if minMessageSize > int(p._getConn().GetMaxMessageSize()) {
+		minMessageSize = len(p.batchBuilder.Compress(payload))
+	}
+
+	if minMessageSize > int(p._getConn().GetMaxMessageSize()) {
```

Comment:
I think a short-circuit `and` will work here
```suggestion
	if len(payload) > int(p._getConn().GetMaxMessageSize()) && 
		len(p.batchBuilder.Compress(payload)) > int(p._getConn().GetMaxMessageSize()) {
```